app/Service/StudentDetailsService.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Service;
+
+use App\Exceptions\StudentNotFoundException;
+use App\Models\Resume;
+use App\Models\Student;
+use App\Models\Tag;
+
+class StudentDetailsService
+{
+    public function execute($student)
+    {
+        return $this->getStudentDetailsById($student);
+    }
+    
+
+    public function getStudentDetailsById($student){
+        
+        $studentDetails = Resume::where('student_id', $student)->first();
+
+        if(!$studentDetails){
+            throw new StudentNotFoundException($student);
```

Comment:
**This is just an opinion, because I myself still don't quite understand how to handle this type of situations:**

In this case, I believe the use of this exception might be a bit confusing, as in the event that a resume doesn't exist, it would throw a message indicating that the student hasn't been found, which might not necessarily be the case.

Here, I would recommend replacing this exception with one that already exists called ``ResumeNotFoundException``.

Reply:
I think you're right, I also think what you propose is more correct.I think you're right, I also think what you propose is more correct.

Reply:
I think you're right, I also think what you propose is more correct.